### DISCUSSION TITLE
I want to Help!

### DISCUSSION THREAD

**Discussion by tris203** [tris203/rzls.nvim#1] [Announcements]
If you want to help with the integration, then leave some information below with your level of experience with Neovim plugins, LSP and C# and we can work out how we can work best on this together

Thanks in advance

**Comment by tris203** [comment]
I have also create a Discord server

https://discord.gg/d652C8bgSR

if anyone is more comfortable there

**Comment by asherjvitek** [comment]
Hello, I came onto this and have been using it. I think that I could help a little. I was working on the "--TODO: Extend this to cover all razor highlights" that is in rzls.nvim\lua\rzls\razor.lua. I think I got all the regex ones and a few other small ones that were pretty easy working.

I know a little about neovim plugins, very little about LSPs but I am learning. I work full time on C# applications and have for a while. If you would like some help with this I am happy to contribute when and where I can.

If you would like I can fork this and then send some PRs when I am done that way or if there is another way you would like help I can do that.

Thanks for the wonderful plugin and work that you have done so far.

**Reply by tris203** [reply]
That would be great. Always happy to review PRs or discuss approaches in issues!

**Comment by icalvo** [comment]
Hello! First of all, thanks for all your effort, it's very much appreciated.

I have started to get into the Vim world just a few months ago, first with IdeaVim for Rider and now I am trying as hard as possible to get everything I need working in Neovim. Regarding C#, I've been working with .NET since version 1 (so 23 years ago), with VB.NET and 12 years ago I switched to C# and never looked back.

I've been trying to make rzls.nvim work with Razor pages (`.cshtml` Razor files) and it looks like it's still has some issues. I want to use `vim.lsp.buf.definition()` and `vim.lsp.buf.references()` to navigate inside the `.cshtml` and especially between the `.cshtml` and its corresponding `.cshtml.cs` view model. Now I have seen it working but under very special conditions:

- If I open neovim in a .NET solution folder and then open a `.cshtml` file, it manages to start Roslyn properly and I can navigate back and forth between the razor file and the viewmodel file. I see "connected to Roslyn via pipe" notification from rzls.nvim.
- If a session manager (e.g. resession.nvim) opens the `.cshtml` file for me, it doesn't work anymore. It has to be manually opened. I see also "connected to Roslyn via pipe" but `vim.lsp.buf.definition()` cannot navigate to the `.cshtml.cs` file.
- If I open manually the `.cshtml.cs` first and try to go from there to the `.cshtml` file using `vim.lsp.buf.references()`, it takes me to the code generated file instead (`_cshtml.g.cs`).

To simplify I am just using the project generated by:

```
dotnet new razor -n web
dotnet new sln
dotnet sln add web
```
I am trying to open the `Error.cshtml` / `Error.cshtml.cs` files and navigate between them using `definition()`/`references()` on the `ShowRequestId` property.
